projects/vdk-core/tests/vdk/internal/builtin_plugins/ingestion/test_ingester_base.py

```diff
@@ -27,12 +26,21 @@ def create_ingester_base() -> IngesterBase:
     }
     test_config = Configuration(None, config_key_value_pairs, {})
 
-    return IngesterBase(
-        data_job_name="test_job",
-        op_id="42a420",
-        ingester=MagicMock(spec=IIngesterPlugin),
-        ingest_config=IngesterConfiguration(test_config),
-    )
+    if kwargs:
+        return IngesterBase(
+            data_job_name="test_job",
+            op_id="42a420",
+            ingester=MagicMock(spec=IIngesterPlugin),
+            ingest_config=IngesterConfiguration(test_config),
+            **kwargs
+        )
+    else:
+        return IngesterBase(
+            data_job_name="test_job",
+            op_id="42a420",
+            ingester=MagicMock(spec=IIngesterPlugin),
+            ingest_config=IngesterConfiguration(test_config),
+        )
```

Comment:
interesting; is it possible to optimise like
if in favour of None:
```def create_ingester_base(kwargs={}) -> IngesterBase:```

then
```suggestion
    return IngesterBase(
        data_job_name="test_job",
        op_id="42a420",
        ingester=MagicMock(spec=IIngesterPlugin),
        ingest_config=IngesterConfiguration(test_config),
        **kwargs
    )
```

Reply:
Setting `kwargs={}` in the method signature was causing issues, as dictionaries are mutable objects, but with a slight workaround it was indeed possible to get rid of the duplicate code. Thank you for the suggestion